ui/analyse/src/autoShape.ts

```diff
@@ -165,72 +143,40 @@ export function compute(ctrl: AnalyseCtrl): DrawShape[] {
-</g>
-`),
-
-  // Interesting move
-  '!?': prependDropShadow(`
-<g transform="translate(77 -18) scale(0.4)">
-  <circle style="fill:#ea45d8;filter:url(#shadow)" cx="50" cy="50" r="50"/>
-    <path fill="#fff" d="M60.823 58.9q0-4.098 1.72-6.883 1.721-2.786 5.9-5.818 3.687-2.622 5.243-4.506 1.64-1.966 1.64-4.588t-1.967-3.933q-1.885-1.393-5.326-1.393t-6.8 1.065q-3.36 1.065-6.883 2.868l-4.343-8.767q4.015-2.212 8.685-3.605 4.67-1.393 10.242-1.393 8.521 0 13.192 4.097 4.752 4.096 4.752 10.405 0 3.36-1.065 5.818-1.066 2.458-3.196 4.588-2.13 2.048-5.326 4.424-2.376 1.72-3.687 2.95-1.31 1.229-1.802 2.376-.41 1.147-.41 2.868v2.376h-10.57zm-1.311 16.632q0-3.77 2.048-5.244 2.049-1.557 4.998-1.557 2.868 0 4.916 1.557 2.049 1.475 2.049 5.244 0 3.605-2.049 5.244-2.048 1.556-4.916 1.556-2.95 0-4.998-1.556-2.048-1.64-2.048-5.244zM36.967 61.849h-9.75l-2.049-39.083h13.847zM25.004 75.532q0-3.77 2.049-5.244 2.048-1.557 4.998-1.557 2.867 0 4.916 1.557 2.048 1.475 2.048 5.244 0 3.605-2.048 5.244-2.049 1.556-4.916 1.556-2.95 0-4.998-1.556-2.049-1.64-2.049-5.244z" vector-effect="non-scaling-stroke"/>
-</g>
-`),
-
-  // Good move
-  '!': prependDropShadow(`
-<g transform="translate(77 -18) scale(0.4)">
-  <circle style="fill:#22ac38;filter:url(#shadow)" cx="50" cy="50" r="50" />
-  <path fill="#fff" d="M54.967 62.349h-9.75l-2.049-39.083h13.847zM43.004 76.032q0-3.77 2.049-5.244 2.048-1.557 4.998-1.557 2.867 0 4.916 1.557 2.048 1.475 2.048 5.244 0 3.605-2.048 5.244-2.049 1.556-4.916 1.556-2.95 0-4.998-1.556-2.049-1.64-2.049-5.244z" vector-effect="non-scaling-stroke"/>
-</g>
-`),
-
-  // Brilliant move
-  '!!': prependDropShadow(`
-<g transform="translate(77 -18) scale(0.4)">
-  <circle style="fill:#168226;filter:url(#shadow)" cx="50" cy="50" r="50" />
-  <path fill="#fff" d="M71.967 62.349h-9.75l-2.049-39.083h13.847zM60.004 76.032q0-3.77 2.049-5.244 2.048-1.557 4.998-1.557 2.867 0 4.916 1.557 2.048 1.475 2.048 5.244 0 3.605-2.048 5.244-2.049 1.556-4.916 1.556-2.95 0-4.998-1.556-2.049-1.64-2.049-5.244zM37.967 62.349h-9.75l-2.049-39.083h13.847zM26.004 76.032q0-3.77 2.049-5.244 2.048-1.557 4.998-1.557 2.867 0 4.916 1.557 2.048 1.475 2.048 5.244 0 3.605-2.048 5.244-2.049 1.556-4.916 1.556-2.95 0-4.998-1.556-2.049-1.64-2.049-5.244z" vector-effect="non-scaling-stroke"/>
-</g>
-`),
+const glyphColors: { [k: string]: string } = {
+  '??': '#df5353',
+  '?': '#e69f00',
+  '?!': '#56b4e9',
+  '!': '#22ac38',
+  '!!': '#168226',
+  '!?': '#ea45d8',
 };
+
+function annotationShapes(ctrl: AnalyseCtrl): DrawShape[] {
+  const shapes: DrawShape[] = [];
+  const { uci, glyphs, san } = ctrl.node;
+  if (ctrl.showMoveAnnotation() && uci && san && glyphs && glyphs.length > 0) {
+    const move = parseUci(uci)!;
+    const destSquare = san.startsWith('O-O') // castle, short or long
+      ? squareRank(move.to) === 0 // white castle
+        ? san.startsWith('O-O-O')
```

Comment:
The purpose is to adjust the arrow for castling moves. With this addition, it always points to the destination of the king, rather than the matching rook if `uci` is in Chess960 notation. It's independent of the variant.